app/models/base_aggregated_cash_transaction.rb

```diff
@@ -14,6 +14,15 @@ class BaseAggregatedCashTransaction # rubocop:disable Metrics/ClassLength
   validate :validate_at_least_one_selected
   validate :checkbox_integrity
 
+  attr_accessor :month1, :month2, :month3
+
+  def initialize
+    super
+    @month1 = Date.today.beginning_of_month - 1.month
+    @month2 = month1 - 1.month
+    @month3 = month2 - 1.month
```

Comment:
This might be preferable if we are ensuring to use the MONTH_RANGE and number of fields like we do elsewhere for consistency.
```
MONTH_RANGE.each do |i|
    instance_variable_set("@month#{i}", Date.today.beginning_of_month - i.month)
end
```